internal/cephfs/fsjournal.go

```diff
@@ -69,18 +82,28 @@ func checkVolExists(ctx context.Context, volOptions *volumeOptions, secret map[s
 	}
 	imageUUID := imageData.ImageUUID
 	vid.FsSubvolName = imageData.ImageAttributes.ImageName
+	var evnf ErrVolumeNotFound
+
+	clone, cloneInfoErr := getcloneInfo(ctx, volOptions, cr, volumeID(vid.FsSubvolName))
+	if cloneInfoErr != nil {
+		if !errors.As(cloneInfoErr, &evnf) {
+			return nil, status.Error(codes.Internal, cloneInfoErr.Error())
+		}
+	} else {
+		if clone.Status.State != cephFSCloneCompleted {
+			return nil, ErrCloneInProgress{err: fmt.Errorf("clone is in progress for %v", vid.FsSubvolName)}
+		}
+	}
 
 	_, err = getVolumeRootPathCeph(ctx, volOptions, cr, volumeID(vid.FsSubvolName))
```

Comment:
This is applicable for non-clone volumes only and serves as an existence check for the subvolume. Need not be done for a clone, as that is covered with the info command above.